### PR TITLE
[MAINT] check compatibility of  (multi)nifti masker with scikitlearn 

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none but relates to https://github.com/nilearn/nilearn/issues/4538#issuecomment-2527259857

Changes proposed in this pull request:

- check compatibility of nifti masker with scikitlearn
- update nifti masker constructor to avoid doing any logic in it: all checks are done at fit time

### CODE REVIEW

File: nilearn/maskers/multi_nifti_masker.py [hunk 1]
```diff
@@ -167,8 +167,6 @@ def __init__(
         verbose=0,
         **kwargs,
     ):
-        if memory is None:
-            memory = Memory(location=None)
```
Comment (Remi-Gau) [inline nilearn/maskers/multi_nifti_masker.py:171]: done in nifti masker

File: nilearn/maskers/multi_nifti_masker.py [hunk 2]
```diff
@@ -185,9 +183,6 @@ def __init__(
             mask_strategy=mask_strategy,
             mask_args=mask_args,
             dtype=dtype,
-            clean_kwargs={
-                k[7:]: v for k, v in kwargs.items() if k.startswith("clean__")
-            },
```
Comment (Remi-Gau) [inline nilearn/maskers/multi_nifti_masker.py:190]: passed via kwargs to nifti masker

File: nilearn/maskers/nifti_masker.py [hunk 3]
```diff
@@ -210,12 +220,20 @@ class NiftiMasker(BaseMasker):
...
     %(memory_level1)s
+
     %(verbose0)s
+
     reports : :obj:`bool`, default=True
         If set to True, data is saved in order to produce a report.
 
+    %(cmap)s
+        default="CMRmap_r"
+        Only relevant for the report figures.
```
Comment (Remi-Gau) [inline nilearn/maskers/nifti_masker.py:235]: I believe it is OK to make this explicit in the API

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Review by man-shu — APPROVED** [review]
LGTM!

**Pull request merged**